source/UIAHandler/utils.py

```diff
@@ -305,9 +305,7 @@ def _shouldUseUIAConsole(hwnd: int) -> bool:
 	else:
 		# #7497: the UIA implementation in old conhost is incomplete, therefore we
 		# should ignore it.
-		# When the UIA implementation is improved, the below line will be replaced
-		# with a check that _getConhostAPILevel >= FORMATTED.
-		return False
+		return _getConhostAPILevel(hwnd) >= WinConsoleAPILevel.FORMATTED

```

Comment:
@seanbudd Some of these changes are dependant on one another: the user guide explains the new default behaviour (ternary option, the fact that review commands don't apply to `FORMATTED` consoles), and making the GUI option ternary without changing the meaning of `auto` would mean that `auto` and `legacy` are equivalent (at least making the label of the option confusing if nothing else). This PR should therefore be taken as "all or nothing" (see #10684 for reversion of some of this code when it was decided to not use UIA by default in what are now called `END_EXCLUSIVE` consoles). Note that the option in the config spec has always been ternary: ticking the box sets `winConsoleImplementation=UIA`, and unticking sets `winConsoleImplementation=auto`. (this was done so that user experimentation during the development of the feature would have no bearing on its eventual switch to default).

Reply:
@codeofdusk - We need a separate PR that just introduces the ternary option in a minimal way.
That way, if we revert the PR, we don't have to revert the ternary option back to a binary checkbox.
It would have been ideal to introduce the option as a ternary option to begin with.